NavigationReactNative/src/cpp/react/renderer/components/navigationreactnative/NVTabBarItemShadowNode.cpp

```diff
@@ -26,11 +27,23 @@ void NVTabBarItemShadowNode::updateStateIfNeeded() {
 
   ensureUnsealed();
 
-  auto state = NVTabBarItemState{
-      newImageSource,
-      imageManager_->requestImage(newImageSource, getSurfaceId()),
-    };
-  setStateData(std::move(state));
+  bool isSystemImageResult = isSystemImage(newImageSource.uri);
+    
+  if (isSystemImageResult) {
+    auto state = NVTabBarItemState{
+        newImageSource,
+        {newImageSource, nullptr, {}}
```

Comment:
I was having some second thoughts about this, wondering whether your constructor overload was a better idea. But then I noticed that the [ShadowNode.h is already using this approach to initialise the state data](https://github.com/grahammendick/navigation/blob/2a37da88def53d431696a1fed153a5ba51ee5072/NavigationReactNative/src/cpp/react/renderer/components/navigationreactnative/NVTabBarItemShadowNode.h#L33) and I feel much better about it now. Great work!